velox/expression/tests/ExpressionFuzzer.h

```diff
@@ -337,41 +319,61 @@ class ExpressionFuzzer {
   // --assign_function_tickets startup flag .
   std::unordered_map<std::string, int> functionsToTickets_;
 
-  /// The remaining levels of expression nesting. It's initialized by
-  /// FLAGS_max_level_of_nesting and updated in generateExpression(). When its
-  /// value decreases to 0, we don't generate subexpressions anymore.
-  int32_t remainingLevelOfNesting_;
-
   /// We allow the arg generation routine to be specialized for particular
   /// functions. This map stores the mapping between function name and the
   /// overridden method.
   using ArgsOverrideFunc = std::function<std::vector<core::TypedExprPtr>(
       const CallableSignature& input)>;
+
   std::unordered_map<std::string, ArgsOverrideFunc> funcArgOverrides_;
 
-  std::shared_ptr<core::QueryCtx> queryCtx_{std::make_shared<core::QueryCtx>()};
-  std::shared_ptr<memory::MemoryPool> pool_{memory::addDefaultLeafMemoryPool()};
-  core::ExecCtx execCtx_{pool_.get(), queryCtx_.get()};
-  test::ExpressionVerifier verifier_;
+  // This pool is needed to generate vectors when constant expressions get
+  // generated.
+  std::shared_ptr<memory::MemoryPool> pool_;
 
-  test::VectorMaker vectorMaker_{execCtx_.pool()};
   VectorFuzzer vectorFuzzer_;
 
+  FuzzerGenerator rng_;
+
   /// Contains the input column references that need to be generated for one
   /// particular iteration.
-  std::vector<TypePtr> inputRowTypes_;
-  std::vector<std::string> inputRowNames_;
+  struct State {
+    void reset() {
+      inputRowTypes_.clear();
+      inputRowNames_.clear();
+      typeToColumnNames_.clear();
+      expressionBank_.reset();
+      expressionStats_.clear();
```

Comment:
Why does expressionStats_ belong to per-iteration state? I thought expressionStats_ should keep the statistics across iterations.

Reply:
The ExpressionFuzzerVerifier have a stats for the total run, but this expressionStats_ only for one iteration 
then the ExpressionFuzzerVerifier uses that to accumulate counting at its side. Other users might not. 

line 503 in ExpressionFuzzerVerifier.cpp
```
   for (auto& [funcName, count] : selectionStats) {
      exprNameToStats_[funcName].numTimesSelected += count;
    }
  ````